frontend/src/metabase/containers/SaveQuestionModal/context.tsx

```diff
@@ -0,0 +1,121 @@
+import {
+  createContext,
+  type PropsWithChildren,
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+} from "react";
+
+import { useListCollectionsQuery } from "metabase/api";
+import { SAVE_QUESTION_SCHEMA } from "metabase/containers/SaveQuestionModal/schema";
+import type {
+  FormValues,
+  SaveQuestionProps,
+} from "metabase/containers/SaveQuestionModal/types";
+import {
+  getInitialValues,
+  getPlaceholder,
+  getTitle,
+  submitQuestion,
+} from "metabase/containers/SaveQuestionModal/util";
+import { FormProvider } from "metabase/forms";
+import { useSelector } from "metabase/lib/redux";
+import { getIsSavedQuestionChanged } from "metabase/query_builder/selectors";
+import type Question from "metabase-lib/v1/Question";
+
+type SaveQuestionContextType = {
+  question: Question;
+  originalQuestion: Question | null;
+  title: string;
+  nameInputPlaceholder: string;
+  initialValues: FormValues;
+  handleSubmit: (details: FormValues) => Promise<void>;
+  values: FormValues;
+  setValues: (values: FormValues) => void;
+  showSaveType: boolean;
+};
```

Comment:
Based on the above comment, I would suggest lifting values that can be derived out of the context. I can spot `nameInputPlaceholder` at first glance, but there might be more we can lift out.

My criteria would be if it doesn't need to be shared across two components or accessed by the SDK user as part of the API, then we don't need it in the context yet. We can put them here later when we need to either share it between components, or expose it as part of the public SDK (or as part of SDK components that would consume the state)

PS. At first I think we could get by without context by passing the props to each component, but if you plan on SDK components consuming the context in the upcoming PR, then this is fine.